fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java

```diff
@@ -3273,8 +3279,8 @@ private void createTaskForMaterializedView(String dbName, MaterializedView mater
             TaskManager taskManager = GlobalStateMgr.getCurrentState().getTaskManager();
             taskManager.createTask(task, false);
             // for event triggered type, run task
-            if (task.getType() == Constants.TaskType.EVENT_TRIGGERED ||
-                    refreshMoment.equals(MaterializedView.RefreshMoment.IMMEDIATE)) {
+            if (task.getType() == Constants.TaskType.EVENT_TRIGGERED &&
```

Comment:
Will this alter any behavior?

Reply:
Before that, the behavior is not correct. Since the  `REFRESH ASYNC` is `EVENT_TRIGGERED`, so it would refresh the materialized view after creating.

After that, the mv would be refreshed if you don't specify the `DEFERRED` keyword, no matter it's `ASYNC` or `MANUAL`.